pheps/phep-0001.md

```diff
@@ -0,0 +1,360 @@
+
+## PHEP Review & Resolution
+<a name="phep-review-resolution"></a>
+Once the authors have completed a PHEP, they may request a review for style and consistency from the PHEP editors. However, content review and acceptance of the PHEP is the responsibility of the community. It should be clear from the [discussion](#discussing-a-phep) when the community is beginning to agree: concerns have been addressed and new ones are not being raised, PR reviews with "approve" status are appearing, discussion is quieting down. At this point the PHEP author should seek formal acceptance.
+
+A PHEP is accepted by consensus of attendees (virtual or in-person) at both a PyHC regular telecon and a regular PyHC meeting, with no edits to the PHEP occurring in between. Potential acceptance of a PHEP must be placed on the agenda at least a month before the meeting, including a link to the open PR with the proposed-final language of the PHEP. Persons not in attendance at the meetings may object via comment on the PR. The PHEP may be edited up until the initial vote, for instance to incorporate discussion at the meeting. However, any edits after a vote invalidate that acceptance, and two further votes (one at a meeting, one on a telecon) must be taken.
+
+A "regular PyHC meeting" is any meeting (virtual, physical, or hybrid) organized for the purpose of discussing PyHC business, of at least a half-day duration, where a substantial portion of the community is expected to participate, and announced as such well in advance. This includes the semiannual meetings but may also include side gatherings at other meetings which are of interest to the community and designated as PyHC meetings.
+
+"Consensus" here means no concrete objections have been raised: an accepted PHEP must be one that the entire community can at least live with. The PHEP authors must make the case for the PHEP; objectors must provide meaningful concerns that could, in principle, be addressed by the authors. All are expected to discuss in good faith, while recognizing that good faith does not guarantee consensus can be reached.
+
+For a PHEP to be accepted it must meet certain minimum criteria. It must be a clear and complete description of the proposed change. The change must represent a net improvement. The proposed implementation, if applicable, must be solid and must not complicate PyHC projects unduly. Finally, a proposed enhancement must be "pythonic" in order to be accepted by the community. ("Pythonic" is an imprecise term; it should be understood in the context of the broad Python ecosystem.)
+
+Once a PHEP has been accepted, the reference implementation, if applicable, must be completed. When the reference implementation is complete and incorporated into a PyHC project's source code repository, the status will be changed to "Final". PHEPs which do not require a reference implementation move immediately to "Final" upon acceptance.
```

Comment:
I suggest we remove the "Accepted" state and require the reference implementation to be complete before acceptance (which will then go immediately to "Final"). So the implementation will be against a technically Draft but hopefully mature standard.

The process here feels a bit weird because there's no provision for how the move from Accepted to Final is evaluated. If it's just the completion of the reference implementation, that seems straightforward (although who agrees that the reference implementation is truly complete/relevant?) But there's also the provision that the PHEP should be updated with lessons learned during implementation--which suggests potentially substantive changes that should be reviewed / approved / voted on.

The downside I can see is somebody going to a huge effort to make a reference implementation for a standard that then gets voted down, or requires major changes to be accepted. I think this can be avoided by getting substantial input before starting that reference implementation. Then the standard and implementation can proceed hand-over-hand however seems appropriate for that PHEP. It seems like a much smaller problem for PyHC than it would for the Python language, where the implementation is probably a _lot_ more work and subject to a lot of possible time-wasting.

Reply:
I agree with this suggestion. The change would simplify the process and I buy the reasoning. Substantial input before starting a reference implementation should prevent wasted effort.